qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -0,0 +1,346 @@
+                this will default to the physical qubits of the experiment.
+            sched_name: The name of the schedule to retrieve from the instance of
+                :class:`Calibrations` stored under the calibration options. If this is None then
+                :meth:`get_schedule_from_calibrations` will default to the :code:`schedule_name`
+                in the calibration options.
+            option_name: The name of the option under which to get the schedule from the experiment
+                options. This will default to "schedule" if None is given.
+            assign_params: A dict that :meth:`get_schedule_from_calibrations` can use to leave
+                certain parameters in the schedule unassigned. The key is the name of the parameter
+                and the value should be an instance of :class:`ParameterExpression`.
+            kwargs: Additional keyword arguments that can be used by implementations of
+                :meth:`get_schedule_from_defaults`.
+
+        Returns:
+            schedules: The schedules (possibly with one or more free parameters) as either a
+                ScheduleBlock or a list of ScheduleBlocks depending on the experiment.
+
+        Raises:
+            CalibrationError: if none of the methods above returned schedules.
+        """
+        schedules = self.get_schedule_from_options(option_name)
+
+        if schedules is None:
+            schedules = self.get_schedule_from_calibrations(qubits, sched_name, assign_params)
+
+        if schedules is None:
+            schedules = self.get_schedule_from_defaults(**kwargs)
+
+        self.validate_schedule(schedules)
+
+        return schedules
+
+    def circuit_metadata(self, xval: Any, **kwargs) -> Dict[str, Any]:
+        """Return the circuit metadata for the calibration experiment."""
+        metadata = {"experiment_type": self._type, "qubits": self.physical_qubits, "xval": xval}
+        metadata.update(kwargs)
+        return metadata
+
+    @classmethod
+    def _default_calibration_options(cls) -> Options:
+        """Default calibration options for the experiment.
+
+        Calibration Options:
+            calibrations (Calibrations): An optional instance of :class:`Calibrations` if this
+                instance is specified then the experiment will try and update the calibrations.
+            auto_update (bool): A boolean which defaults to True. If this variable is set to
+                True then running the calibration experiment will block for the results and
+                update the calibrations if the calibrations is not None.
+            schedule_name (str): The name of the schedule to retrieve from the calibrations.
+            cal_parameter_name (str): The name of the parameter to update in the calibrations.
+        """
+        return Options(
+            calibrations=None, auto_update=True, schedule_name=None, cal_parameter_name=None
```

Comment:
Are these all necessary as options that can be changed after a calibration experiment is initialized, or should some be init args of the base cal experiment that can't be changed after initialization? In particular it seems like `calibrations` is a good candidate for something that should be fixed during init and not changed.

If the same with schedule_name and cal_parameter_name, im not sure we need a separate options method, since subclasses can just use regular experiment options for any additional ones.

The only possible one to keep is auto_update which we could just make a separate setter for the base cal experiment if its the only extra one.

Reply:
This makes sense. This also means that a given experiment instance will always be running the same schedule and parameter update. This is probably a good thing and will keep the resulting experiment data consistent. See 5c2d9c1 and d354708